src/Plugin/AccessControlHierarchy/Menu.php

```diff
@@ -147,8 +147,16 @@ public function alterForm(AccessSchemeInterface $scheme, array &$form, FormState
     foreach ($element['link']['menu_parent']['#options'] as $id => $data) {
       // The menu value here prepends the menu name. Remove that.
       $parts = explode(':', $id);
-      $menu = array_shift($parts);
-      $sections = [implode(':', $parts)];
+      $menu = $parts[0];
+      // If the second element is empty, this is the root element which is
+      // checked by menu name.
+      if (empty($parts[1])) {
+        $sections = [$menu];
+      }
+      else {
+        array_shift($parts);
```

Comment:
would it be better to do this above instead of using `$parts[0]`, you could then use `if ($parts)` instead of `if(empty($parts[1]))`

Reply:
Maybe. I was doing that before.